tooling/minimize/src/rvalue.rs

```diff
@@ -250,6 +250,32 @@ impl<'cx, 'tcx> FnCtxt<'cx, 'tcx> {
                             );
                         }
                     }
+                    smir::CastKind::PointerCoercion(smir::PointerCoercion::Unsize) => {
+                        let operand_ty = operand.ty(&self.locals_smir).unwrap();
+                        let old_pointee_ty = self.translate_pointee_ty_smir(operand_ty, span);
+                        let new_pointee_ty = self.translate_pointee_ty_smir(*ty, span);
+                        let Type::Ptr(new_ptr_ty) = self.translate_ty_smir(*ty, span) else {
+                            rs::span_bug!(span, "ptr to ptr cast to non-pointer");
+                        };
+                        let operand = self.translate_operand_smir(operand, span);
+                        match (old_pointee_ty, new_pointee_ty) {
+                            (Type::Array { count, elem: a_elem }, Type::Slice { elem: s_elem }) => {
+                                if a_elem != s_elem {
+                                    rs::span_bug!(
+                                        span,
+                                        "Unsizing to slice with different element type"
+                                    );
+                                }
+                                // FIXME(UnsizedTypes): do we support 32bit architectures here? If so, `usize` is probably wrong.
+                                build::construct_wide_pointer(
+                                    operand,
+                                    build::const_int_typed::<usize>(count),
+                                    Type::Ptr(new_ptr_ty),
+                                )
+                            }
+                            _ => rs::span_bug!(span, "Unsupported unsizing coercion"),
```

Comment:
Please say in the error from which type to which type the unsupported coercion goes.